src/rpdk/core/generate_canary.py

```diff
@@ -0,0 +1,83 @@
+import json
+import logging
+from pathlib import Path
+
+from .generate_stack_templates import StackTemplateGenerator
+
+TARGET_CANARY_ROOT_FOLDER = "canary-bundle"
+TARGET_CANARY_FOLDER = "canary-bundle/canary"
+RPDK_CONFIG_FILE = ".rpdk-config"
+CANARY_FILE_PREFIX = "canary"
+CONTRACT_TEST_DEPENDENCY_FILE_NAME = "dependencies.yml"
+CANARY_DEPENDENCY_FILE_NAME = "bootstrap.yaml"
+CANARY_SETTINGS = "canarySettings"
+TYPE_NAME = "typeName"
+CONTRACT_TEST_FILE_NAMES = "contract_test_file_names"
+INPUT1_FILE_NAME = "inputs_1.json"
+FILE_GENERATION_ENABLED = "file_generation_enabled"
+
+
+class CanaryFileGenerator:
+    def __init__(self, root=None):
+        self.root = Path(root) if root else Path.cwd()
+        self.canary_settings = {}
+        self.type_name = ""
+        self.contract_test_file_names = [INPUT1_FILE_NAME]
+        self.load_rpdk_config()
+
+    def load_rpdk_config(self) -> None:
+        """
+        Load the RPDK configuration file.
+
+        This method loads the RPDK configuration file from the current working directory.
+        If the file is not found, it logs a warning and returns an empty dictionary.
+        """
+        config_file_path = self.rpdk_config
+        if not config_file_path.is_file():
+            logging.warning("RPDK configuration file not found")
+            self.canary_settings = {}
+            return
+        with open(config_file_path, "r", encoding="utf-8") as f:
+            rpdk_config = json.load(f)
+            self.canary_settings = rpdk_config.get(CANARY_SETTINGS, {})
+            self.type_name = rpdk_config.get(TYPE_NAME, "")
+            self.contract_test_file_names = self.canary_settings.get(
+                CONTRACT_TEST_FILE_NAMES, [INPUT1_FILE_NAME]
+            )
```

Comment:
dont we already have this method in project? https://github.com/aws-cloudformation/cloudformation-cli/blob/4a8d9a0a92b97c190724eeee2026b55c4cd48b58/src/rpdk/core/project.py#L215, can you reuse that?

Reply:
We have the flexibility to approach this in different ways. However, my preference is to load canary settings separately, allowing it to function as an independent component. By doing so, we can logically separate both components, resulting in a cleaner and more organized codebase.